File: 总题库/206.ReverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        tail = None
        while head != None:
            temp = head
            head = head.next
            temp.next = tail
            tail = temp
        
        return tail

File: 总题库/206.ReverseLinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* tail = NULL;
        while (head != NULL) {
            ListNode* temp = head;
            head = head->next;
            temp->next = tail;
            tail = temp;
        }

        return tail;
    }
};

// 1 2 3 4 5 NULL